### PR TITLE
[AC-7873] Create office hours in blocks

### PR DESCRIPTION
**Pull request by jonkiparsky**
https://masschallenge.atlassian.net/browse/AC-7873

Verify that 
- blocks >= 30 minutes make appropriate office hours
- blocks < 30 minutes make no office hours
- creation/edit behaves sanely

### CODE REVIEW

File: web/impact/impact/v1/views/office_hour_view.py [hunk 1]
```diff
@@ -112,13 +115,21 @@ def handle_response(self, request):
...
+        invalid_serializers = [s for s in serializers if not s.is_valid()]
+        if invalid_serializers:
+            return handle_fail(invalid_serializers[0].errors)
+        for serializer in serializers:
+            self.perform_create(serializer)
+        office_hour = serializers[0].instance
+        if request.user != office_hour.mentor:
+            self.handle_send_mail(office_hour)
+
+        return handle_success(serializers[0].data)
```
Comment (iankigen) [inline web/impact/impact/v1/views/office_hour_view.py]: Non-blocker: Should we return every office hour in the block or just the first one?

**Review by iankigen — APPROVED** [review]
Code looks good.
QA:
- API modified to segment office hours into 30-minute sessions ✅ 
- splits office hour block period into as many complete individual 30-minute office hour sessions ✅
- returns error when office hour block period cannot be segmented into at least 1 30-minute office hour session ✅

File: web/impact/impact/v1/views/office_hour_view.py (hunk 1 above)
Comment (jonkiparsky) [reply web/impact/impact/v1/views/office_hour_view.py]: Interesting question. Depends what the front-end needs to have to do what it needs to do. 

I'm not super familiar with the requirements here, happy to make changes as needed.

Comment (iankigen) [reply web/impact/impact/v1/views/office_hour_view.py]: @PrinceDavis what do you think of this?

Comment (PrinceDavis) [reply web/impact/impact/v1/views/office_hour_view.py]: return all sessions is the expectation, thanks @iankigen for highlighting

Comment (jonkiparsky) [reply web/impact/impact/v1/views/office_hour_view.py]: @PrinceDavis I've made this change. Be aware that this means response.data['data'] is now a list

**Comment by geofrocker** [comment]
I have done QA and it conforms to the done of the ticket and the results highlighted above.

**Review by geofrocker — APPROVED** [review]
Looks good

**Pull request merged**